components/scream/scripts/atm-config-chg

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+"""
+Change a runtime parameter for SCREAM/atm. Run from your case
+after case.setup.
+"""
+
+import argparse, sys, pathlib, os, re
+import xml.etree.ElementTree as ET
+
+###############################################################################
+def expect(condition, error_msg, exc_type=SystemExit, error_prefix="ERROR:"):
+###############################################################################
+    """
+    Similar to assert except doesn't generate an ugly stacktrace. Useful for
+    checking user error, not programming error.
+    """
+    if not condition:
+        msg = error_prefix + " " + error_msg
+        raise exc_type(msg)
+
+###############################################################################
+def change_matches(param_namespace, parent, child_name):
+###############################################################################
+    param_ns_re = re.compile(".*" + param_namespace + "$")
+    m = param_ns_re.match(parent + "::" + child_name)
+    return m is not None
+
+###############################################################################
+def perform_change(element, change, parent, any_change):
```

Comment:
If I understand correctly, if a parameter has 2+ namespaces, say `A::B::param_name`, the change has to be issued as
```
atm-config-chg B::param_name=blah
```
Is that correct? If so, any reason not to force the user to do
```
atm-config-chg A::B::param_name=blah
```
?

Reply:
It will match however many namespaces you provide. "B::param" and "A::B::param" will both match. The fewer namespace qualifications the user provides, the more chance of an ambiguity, which is an error. If there is no ambiguity, then the user can use shorter names to refer to things.